passkeys-backend/assets/index.html

```diff
@@ -0,0 +1,336 @@
+            try {
+                const response = await fetch(`./authentication/start`);
+                const responseJSON = await response.json();
+
+
+                window.webauthnJSON.get(responseJSON)
+                    .then(async (publicKeyCredential) => {
+                        const { id, rawId, response, type, authenticatorAttachment } = publicKeyCredential
+                        const { authenticatorData, clientDataJSON, signature, userHandle } = response
+
+                        const authentication = await fetch('./authentication/verification', {
+                            method: "POST",
+                            headers: {
+                                "Content-Type": "application/json"
+                            },
+                            body: JSON.stringify({
+                                id: id,
+                                rawId: rawId,
+                                clientDataJson: clientDataJSON,
+                                authenticatorData: authenticatorData,
+                                signature: signature,
+                                userHandle: userHandle
+                            })
+                        });
+
+                        const authenticationJSON = await authentication.json();
+                        const {status, identity} = authenticationJSON
+                        if(status === "approved") {
+                            sessionStorage.setItem('session', identity);
+                            loadApp(identity);
+                        } else {
+                            console.log(status);
+                        }
+                    })
+                    .catch((err) => {
+                        if(err) {
+                            console.error("Something goes wrong or maybe you dont have a passkey for this application yet");
+                        }
+                    });
+            } catch (error) {
+                console.log(err);
+            }
+        }
+ 
+        const signUp = async () => {
+            const username = usernameElement.value
+
+            try {
+                const response = await fetch(`./registration/start`, {
+                    method: "POST",
+                    headers: {
+                        "Content-Type": "application/json"
+                    },
+                    body: JSON.stringify({
+                        username: username
+                    })
+                });
+
+                const responseJSON = await response.json();
+                
+                let credential = await window.webauthnJSON.create({publicKey: responseJSON});
+
+
+                const { id, rawId, authenticatorAttachment, response: pubKeyCredResponse, type } = credential;
+                const { attestationObject, clientDataJSON, transports } = pubKeyCredResponse;
+
+                const verificationResponse = await fetch(`./registration/verification`, {
```

Comment:
Do we need to construct this in this way if we're using webauthn-json? 

Can we instead skip lines 302/303 and do the fetch with `body: JSON.stringify(credential);`

I haven't tested this but I think that should work

Reply:
The thing about destructuring here and not sending the credential directly is because the function receive all the data in a flatten object in order to check every property, I did modify these lines to reduce the destructuring but do you think is better to modify the functions to receive the credential directly?